manager/controllers/app/blueprint_controller.go

```diff
@@ -99,7 +99,9 @@ func (r *BlueprintReconciler) reconcileFinalizers(blueprint *app.Blueprint) erro
 				r.Log.V(0).Info("Error while deleting owned resources: " + err.Error())
 				return err
 			}
-
+			if r.hasExternalResources(blueprint) {
+				return errors.NewPlain("helm release uninstall is still in progress")
```

Comment:
Are you relying on exponential backoff here? I think treating this case differently in `Reconcile` and using RequeueAfter would be a better choice.

Reply:
Added a different Result for this case.